service/controller/resource/azureconfig/create.go

```diff
@@ -0,0 +1,472 @@
+	}
+
+	{
+		apiServerDomain, err := newAPIServerDomain(azureCluster)
+		if err != nil {
+			return providerv1alpha1.Cluster{}, microerror.Mask(err)
+		}
+		apiServerIPRange := r.viper.GetString(r.flag.Service.Cluster.Kubernetes.API.ClusterIPRange)
+		if apiServerIPRange == "" {
+			return providerv1alpha1.Cluster{}, microerror.Maskf(executionFailedError, "Kubernetes IP range must not be empty")
+		}
+
+		commonCluster.Kubernetes.API.ClusterIPRange = apiServerIPRange // TODO rename (HOST_SUBNET_RANGE in k8s-vm)
+		commonCluster.Kubernetes.API.Domain = apiServerDomain
+		commonCluster.Kubernetes.API.SecurePort = r.viper.GetInt(r.flag.Service.Cluster.Kubernetes.API.SecurePort)
+	}
+
+	{
+		_, ipNet, err := net.ParseCIDR(commonCluster.Kubernetes.API.ClusterIPRange)
+		ip := ipNet.IP
+		if err != nil {
+			return providerv1alpha1.Cluster{}, microerror.Mask(err)
+		}
+		ip[3] = kubeDNSIPLastOctet
+
+		commonCluster.Kubernetes.DNS.IP = ip.String()
+	}
+
+	{
+		commonCluster.Kubernetes.Domain = r.viper.GetString(r.flag.Service.Cluster.Kubernetes.Domain)
+	}
+
+	{
+		ingressControllerDomain, err := r.newIngressControllerDomain(azureCluster)
+		if err != nil {
+			return providerv1alpha1.Cluster{}, microerror.Mask(err)
+		}
+		ingressWildcardDomain, err := r.newIngressWildcardDomain(azureCluster)
+		if err != nil {
+			return providerv1alpha1.Cluster{}, microerror.Mask(err)
+		}
+
+		commonCluster.Kubernetes.IngressController.Domain = ingressControllerDomain
+		commonCluster.Kubernetes.IngressController.Docker.Image = r.viper.GetString(r.flag.Service.Cluster.Kubernetes.IngressController.Docker.Image)
+		commonCluster.Kubernetes.IngressController.WildcardDomain = ingressWildcardDomain
+		commonCluster.Kubernetes.IngressController.InsecurePort = r.viper.GetInt(r.flag.Service.Cluster.Kubernetes.IngressController.InsecurePort)
+		commonCluster.Kubernetes.IngressController.SecurePort = r.viper.GetInt(r.flag.Service.Cluster.Kubernetes.IngressController.SecurePort)
+	}
+
+	{
+		kubeletDomain, err := newKubeletDomain(azureCluster)
+		if err != nil {
+			return providerv1alpha1.Cluster{}, microerror.Mask(err)
+		}
+
+		commonCluster.Kubernetes.Kubelet.AltNames = r.viper.GetString(r.flag.Service.Cluster.Kubernetes.Kubelet.AltNames)
+		commonCluster.Kubernetes.Kubelet.Domain = kubeletDomain
+		commonCluster.Kubernetes.Kubelet.Labels = r.viper.GetString(r.flag.Service.Cluster.Kubernetes.Kubelet.Labels)
+		commonCluster.Kubernetes.Kubelet.Port = r.viper.GetInt(r.flag.Service.Cluster.Kubernetes.Kubelet.Port)
+	}
+
+	{
+		userList, err := newSpecClusterKubernetesSSHUsers(r.viper.GetString(r.flag.Service.Cluster.Kubernetes.SSH.UserList))
+		if err != nil {
+			return providerv1alpha1.Cluster{}, microerror.Mask(err)
+		}
+
+		commonCluster.Kubernetes.SSH.UserList = userList
+	}
+
+	{
+		commonCluster.Masters = newSpecClusterMasterNodes()
+		commonCluster.Workers = newSpecClusterWorkerNodes()
```

Comment:
Wouldn't this always create 3 worker nodes independently of what the customer has configured? We could pass `len(workers)` as parameters to `newSpecClusterWorkerNodes` so that we know how many to create.